src/Symfony/Component/FeatureFlag/Provider/ChainProvider.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag\Provider;
+
+/**
+ * @experimental
+ */
+final class ChainProvider implements ProviderInterface
+{
+    public function __construct(
+        /** @var list<ProviderInterface> */
+        private readonly iterable $providers = [],
+    ) {
+    }
+
+    public function get(string $featureName): ?\Closure
+    {
+        foreach ($this->providers as $provider) {
```

Comment:
might be out of scope but just in case : we could store a `featureName` <=> `provider`|`Closure`|else to avoid this foreach.

Reply:
It means that we will have to init this mapping with all available features. Not a problem for now with simple features but it could be problematic if the feature list is unknown (ex: with SaaS)

Reply:
no need to init. It could be on the fly like a cache system. to prevent checking multiple provider every time we do a feature check. On default install it won't change a thing but depending on the implementation behind the provider, it could have a significant impact I think

Reply:
agreed. I got confused on the order of classes. My bad.